components/src/core/components/Input/DateInput.vue

```diff
@@ -5,21 +5,29 @@
         :hasError="hasError"
         :disabled="disabled"
         :readonly="readonly"
-        @blur="onBlur"
-        @update:modelValue="onDateTyped"
-        @click="toggleDropdown"
         :value="displayDate"
         :placeholder="placeholder"
         ref="oxdInput"
+        @update:modelValue="onDateTyped"
+        @click="toggleDropdown"
+        @blur="onBlur"
+        @keyup.esc="closeDropdown"
       />
       <oxd-icon
         :class="dateIconClasses"
         name="calendar"
+        tabindex="0"
         @click="toggleDropdown"
+        @keyup.enter.prevent.stop="toggleDropdown"
       />
     </div>
     <transition name="transition-fade-down">
-      <div v-if="open" class="oxd-date-input-calendar">
+      <div
+          v-if="open"
+          class="oxd-date-input-calendar"
+          v-click-outside="onBlur"
```

Comment:
do we have to add this directive ? can't we get away with onBlur from parent or even from this div?

Reply:
Yes I think we can use onBlur here but the issue is Whenever the dropdown opens the autofocus should be shifted to the dropdowns topmost-left element. If that happens , then the onBlur event will work because the focus is in the dropdown. But currently without this directive it will not close the dropdown if we click a whitespace. 

to set the autofocus We have a directive for that in the application side. But Thilanka said that we have to use almost simmilar directive like that in the oxd side with some changes since this is not a modal. It is yet to be implemented I think